src/main/java/org/apache/maven/it/Embedded3xLauncher.java

```diff
@@ -268,4 +271,26 @@ public String getMavenVersion()
         throw new LauncherException( "Could not determine embedded Maven version" );
     }
 
+    /**
+     * Replicates the logic from <a href="https://git.io/JSMug">Maven start script</a> to find 
+     * the project's base directory.
+     * @param workingDirectory the working directory
+     * @return the project's base directory
+     */
+    private static File findProjectBaseDirectory( File workingDirectory )
```

Comment:
Works both on Windows and POSIX-like?

Reply:
yes, this is mostly taken over from m2e: https://git.eclipse.org/c/m2e/m2e-core.git/commit/?id=55e4c7a4afc0ae34a81ce55e9958c341e1a23c04